src/backend/mod.rs

```diff
@@ -85,4 +89,24 @@ impl DNSBackend {
 
         DNSResult::Success(results)
     }
+
+    // reverse lookup must return a single name resolved via mapping
+    pub fn reverse_lookup(&self, requester: &IpAddr, lookup_ip: &IpAddr) -> Vec<String> {
+        let nets = match self.ip_mappings.get(requester) {
+            Some(n) => n,
+            None => return vec![],
+        };
+
+        for net in nets {
+            match self.reverse_mappings.get(net) {
+                Some(n) => match n.get(lookup_ip) {
+                    Some(i) => return i.clone(),
```

Comment:
Is there a reason not to borrow here, i.e. return `Vec<&str>` and have the caller do the clone if they need to?